src/main/java/org/jetbrains/research/kotlincodesmelldetector/KotlinCodeSmellFacade.java

```diff
@@ -54,4 +59,85 @@
         Collections.sort(moveMethodCandidateRefactorings);
         return moveMethodCandidateRefactorings;
     }
+
+    public static Set<ASTSliceGroup> getExtractMethodRefactoringOpportunities(ProjectInfo project, ProgressIndicator indicator) {
+        Set<ASTSliceGroup> extractedSliceGroups = new HashSet<>();
+        FirFile firFile = getCurrentFirFileOpenInEditor(project.getProject());
+        List<FirSimpleFunction> firSimpleFunctions = extractFirSimpleFunctions(firFile);
+        for (FirSimpleFunction firSimpleFunction : firSimpleFunctions) {
+            processMethod(extractedSliceGroups, firSimpleFunction);
+        }
+        return extractedSliceGroups;
+    }
+
+    private static void processMethod(final Set<ASTSliceGroup> extractedSliceGroups, FirSimpleFunction firSimpleFunction) {
```

Comment:
We may do it not only for `FirSimpleFunction`, but for any `FirFunction` with body